test/ppx_import_support/test.ml

```diff
@@ -0,0 +1,102 @@
+(* Test for the ppx_import old syntax compat support *)
+
+open Ppxlib
+
+let id =
+  Extension.__declare_ppx_import
+    "id"
+    (fun ~ctxt:_ td ->
+       match td.ptype_manifest with
+       | Some {ptyp_desc = Ptyp_extension (_, PTyp wrapped_manifest); _} ->
+         {td with ptype_manifest = Some wrapped_manifest}
+       | _ -> assert false)
+[%%expect{|
+val id : Extension.t = <abstr>
+|}]
+
+Driver.register_transformation
+  ~rules:[Context_free.Rule.extension id]
+  "id"
+[%%expect{|
+- : unit = ()
+|}]
+
+(* The expander receives the type decl with the extension point removed, it should preserve
+   attibutes *)
+type t = [%id: int]
+[%%expect{|
+type t = int
+|}]
+
+(* It also should work in signatures by default *)
+module type T = sig
+  type t = [%id: int]
+end
+[%%expect{|
+module type T = sig type t = int end
+|}]
+
+let foo =
+  Deriving.add "foo"
+    ~str_type_decl:(Deriving.Generator.make_noarg
+                      (fun ~loc ~path:_ _ -> [%str let foo = 42]))
+    ~sig_type_decl:(Deriving.Generator.make_noarg
+                      (fun ~loc ~path:_ _ -> [%sig: val foo : int]))
+[%%expect{|
+val foo : Deriving.t = <abstr>
+|}]
+
+(* It should properly compose with [@@deriving] *)
+type t = [%id: int]
+[@@deriving foo]
+[%%expect{|
+type t = int
+val foo : t = 42
+|}]
```

Comment:
How does this test show that `[%id]` composes well with `[@@deriving]`, i.e. that it's applied before? Does it show that by inferring the type of `foo` as `t`, as opposed to `int`?

Reply:
You're right that the test isn't precise enough with regards to the ordering, I'll rewrite the deriver to check that the manifest isn't an extension point!

Reply:
That was a very good catch! It turns out that it is indeed applied AFTER the deriving. I completely misunderstood the traversal here, I'll work on fixing this!